File: backend/src/main/java/io/hb3nce04/keykeeperapp/feature/applicant/ApplicantService.java
package io.hb3nce04.keykeeperapp.feature.applicant;

import org.springframework.stereotype.Service;

import io.hb3nce04.keykeeperapp.common.service.AbstractCrudService;
import io.hb3nce04.keykeeperapp.feature.applicant.model.dto.ApplicantRequestDto;
import io.hb3nce04.keykeeperapp.feature.applicant.model.dto.ApplicantResponseDto;
import io.hb3nce04.keykeeperapp.feature.applicant.model.entity.Applicant;

@Service
public class ApplicantService extends AbstractCrudService<Applicant, ApplicantRequestDto, ApplicantResponseDto, ApplicantRepository, ApplicantMapper> {
    public ApplicantService(
            ApplicantRepository repository,
            ApplicantMapper mapper) {
        super(repository, mapper);
    }
}

File: frontend/src/app/features/dashboard/applicants/applicant.service.ts
import {Injectable} from '@angular/core';
import {AbstractCrudService} from '../../../core/services/abstract-crud.service';
import {ApplicantResponseDto} from './dtos/applicant.response.dto';
import {ApplicantRequestDto} from './dtos/applicant-request.dto';

@Injectable({providedIn: 'root'})
export class ApplicantService extends AbstractCrudService<ApplicantRequestDto, ApplicantResponseDto> {
  constructor() {
    super('/applicants');
  }
}
